private-templates-service/client/src/components/Designer/Designer.tsx

```diff
@@ -54,8 +63,12 @@ class Designer extends React.Component<DesignerProps> {
   constructor(props: DesignerProps) {
     super(props);
     props.setPage(this.props.templateName, "Designer");
+    this.setState({isSaveOpen: false});
   }
 
+  toggleModal = () => {
+    this.setState({isSaveOpen: !this.props.isSaveOpen});
```

Comment:
`this.props.isSaveOpen` is never defined, I think it will do `!undefined` which evaluates to `true` which is why it works but i think this should be 

`this.setState({isSaveOpen: !this.state.isSaveOpen});`